lightning-cd.py

```diff
@@ -79,114 +85,113 @@ def drawFileList(t, ystart, yend, mode, selected):# {{{
                 files = None
-                continue
-            writeText(t, 0, t.height() - 1, searchBuffer, termbox.WHITE, 0)
-        modeText = 'search' if mode == SEARCH else 'normal'
-        writeText(t, 0, 0, modeText + ": ", termbox.WHITE, 0)
-        writeText(t, len(modeText) + 2, 0, os.path.realpath('.'), termbox.WHITE, 0)
-        t.present()
-
-        event = t.poll_event()
-        letter, keycode = event[1], event[2]
-        if keycode == termbox.KEY_SPACE:
-            mode, selected, searchBuffer = switchMode(mode, selected)
-        elif letter == ',':
-            os.chdir('..')
-            searchBuffer = ''
-            if not persistentMode:
-                mode = defaultMode
-            selected = selectedValueForMode(mode)
-            files = None
-        elif letter == ';':
-            runCommandOnFile(os.path.realpath('.'), 'true')
-        elif letter == '\'':
-            mode, selected, searchBuffer = takeActionOnPath(files[0 if mode == SEARCH else selected], os.path.realpath('.'))
-            files = None
-        elif letter == '"':
-            files = None
-        elif mode == NORMAL:
-            if letter == 'k':
-                selected = selected - 1 % len(files)
-            elif letter == 'j':
-                selected = selected + 1 % len(files)
-            elif letter == 'v':
-                runCommandOnFile(os.path.realpath('.'), editor + ' ' + files[selected])
-            elif letter == 'f':
-                runCommandOnFile(os.path.realpath('.'), fileBrowser + ' ' + os.path.realpath('.') + ' > /dev/null 2>&1')
-            elif letter == 't':
-                runCommandOnFile(os.path.realpath('.'), 'tmux > /dev/null')
-        elif mode == SEARCH:
-            if letter:
-                if letter in charRange:
-                    searchBuffer = searchBuffer + letter
-                elif letter == '-':
+            elif mode == Mode.NORMAL:
+                if letter == keybindings.KEY_UP:
+                    selected = (selected - 1) % len(files)
+                elif letter == keybindings.KEY_DOWN:
+                    selected = (selected + 1) % len(files)
+                elif letter == keybindings.KEY_EDITOR:
+                    runCommandOnFile(os.path.realpath('.'), editor + ' ' + files[selected])
+                elif letter == keybindings.KEY_FILE_BROWSER:
+                    runCommandOnFile(os.path.realpath('.'), fileBrowser + ' ' + os.path.realpath('.') + ' > /dev/null 2>&1')
+                elif letter == keybindings.KEY_TMUX:
+                    runCommandOnFile(os.path.realpath('.'), 'tmux > /dev/null')
+            elif mode == Mode.SEARCH:
+                if letter:
+                    if letter in charRange:
+                        searchBuffer = searchBuffer + letter
+                    elif letter == keybindings.KEY_DELETE:
+                        searchBuffer = searchBuffer[:-1]
+                elif keycode == 127: # DELETE
```

Comment:
This, if actually used in the code (I would prefer to just leave it as a user-definable printable character), should probably be termbox.TB_KEY_BACKSPACE2 (there's a list of all constants in termbox.h)